fabtests/functional/multi_ep.c

```diff
@@ -207,19 +228,25 @@ static int do_transfers(void)
 
 static int setup_client_ep(int idx)
 {
-	int ret;
+	int ret, av_bind_idx = idx, cq_bind_idx = idx;
+
+	if (shared_cq)
+		cq_bind_idx = 0;
+
+	if (shared_av)
+		av_bind_idx = 0;
 
 	ret = fi_endpoint(domain, fi, &eps[idx], NULL);
 	if (ret) {
 		FT_PRINTERR("fi_endpoint", ret);
 		return ret;
 	}
 
-	ret = ft_alloc_ep_res(fi, &txcqs[idx], &rxcqs[idx], NULL, NULL, NULL, &av);
+	ret = ft_alloc_ep_res(fi, &txcqs[idx], &rxcqs[idx], NULL, NULL, NULL, &avs[idx]);
```

Comment:
Just to clarify intent here - we still create all CQs but then only bind to the first one if the user wants to share it?

Reply:
Yes, that allows for minimal code change. I can also do the way that only create 1 cq/av when shared, if you prefer that.

Reply:
No need, just wanted to clarify that that was intended. Thanks!